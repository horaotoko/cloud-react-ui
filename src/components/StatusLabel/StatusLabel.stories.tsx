import { text, radios, boolean } from "@storybook/addon-knobs";
import { StatusLabel as StatusLabelComponent, Props } from "./StatusLabel";

export default {
  component: StatusLabelComponent,
  title: "components/Status Label",
};

const colorOptions = {
  red: "red",
  gray: "gray",
  green: "green",
  orange: "orange",
  blue: "blue",
};

export const StatusLabel = (): JSX.Element => (
  <StatusLabelComponent
    color={radios("Color", colorOptions, "gray") as Props["color"]}
    bold={boolean("Bold", false)}
    outline={boolean("Outline", false)}
  >
    {text("Text", "Status")}
  </StatusLabelComponent>
);
