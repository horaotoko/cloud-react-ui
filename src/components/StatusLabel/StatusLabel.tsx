import { forwardRef } from 'react';
import styled, { css } from 'styled-components';
import { defaultProps } from '../../theme';

export type Props = {
  color: 'gray' | 'red' | 'green' | 'orange' | 'blue';
  bold?: boolean;
  outline?: boolean;
  className?: string;
  children: React.ReactNode;
};

const StyledStatusLabel = styled.span<Props>`
  ${({ color, bold = false, outline = false, theme: { statusLabel } }) => css`
    display: ${statusLabel.display};
    height: ${statusLabel.height[outline ? 'outline' : 'default']};
    padding: ${statusLabel.padding[outline ? 'outline' : 'default']};
    line-height: ${statusLabel.lineHeight};
    background-color: ${statusLabel.backgroundColor[
      outline ? 'outline' : color
    ]};
    border-width: ${statusLabel.borderWidth};
    border-style: ${statusLabel.borderStyle};
    border-color: ${statusLabel.borderColor[color]};
    border-radius: ${statusLabel.borderRadius};
    color: ${outline
      ? statusLabel.borderColor[color]
      : statusLabel.color[color]};
    font-size: ${statusLabel.fontSize};
    font-weight: ${statusLabel.fontWeight[bold ? 'bold' : 'normal']};
    box-sizing: border-box;
  `}
`;
StyledStatusLabel.defaultProps = defaultProps;

const StatusLabel = forwardRef<HTMLDivElement, Props>((props, ref) => (
  <StyledStatusLabel ref={ref} {...props} />
));

StatusLabel.displayName = 'StatusLabel';

export { StatusLabel };
