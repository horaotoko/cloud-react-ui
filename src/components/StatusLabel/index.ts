export { StatusLabel } from './StatusLabel';
export type { Props as StatusLabelProps } from './StatusLabel';
