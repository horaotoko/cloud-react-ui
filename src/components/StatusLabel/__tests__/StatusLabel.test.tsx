import { render } from "@testing-library/react";
import { StatusLabel } from "../StatusLabel";

describe("StatusLabel", () => {
  describe("snapshot test", () => {
    it("default", () => {
      const { asFragment } = render(
        <StatusLabel color="red">status label</StatusLabel>
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it("bold=true", () => {
      const { asFragment } = render(
        <StatusLabel color="gray" bold>
          status label(bold)
        </StatusLabel>
      );
      expect(asFragment()).toMatchSnapshot();
    });

    it("outline=true", () => {
      const { asFragment } = render(
        <StatusLabel color="green" outline>
          status label(outline)
        </StatusLabel>
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
