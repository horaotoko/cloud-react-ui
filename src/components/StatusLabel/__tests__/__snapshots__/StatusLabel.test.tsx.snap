// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatusLabel snapshot test bold=true 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  height: 18px;
  padding: 2px 3px;
  line-height: 12px;
  background-color: #999999;
  border-width: 1px;
  border-style: solid;
  border-color: #999999;
  border-radius: 4px;
  color: #FFFFFF;
  font-size: 12px;
  font-weight: bold;
  box-sizing: border-box;
}

<span
    class="c0"
    color="gray"
  >
    status label(bold)
  </span>
</DocumentFragment>
`;

exports[`StatusLabel snapshot test default 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  height: 18px;
  padding: 2px 3px;
  line-height: 12px;
  background-color: #EC4949;
  border-width: 1px;
  border-style: solid;
  border-color: #EC4949;
  border-radius: 4px;
  color: #FFFFFF;
  font-size: 12px;
  font-weight: normal;
  box-sizing: border-box;
}

<span
    class="c0"
    color="red"
  >
    status label
  </span>
</DocumentFragment>
`;

exports[`StatusLabel snapshot test outline=true 1`] = `
<DocumentFragment>
  .c0 {
  display: inline-block;
  height: 20px;
  padding: 3px 6px;
  line-height: 12px;
  background-color: #FFFFFF;
  border-width: 1px;
  border-style: solid;
  border-color: #65AB51;
  border-radius: 4px;
  color: #65AB51;
  font-size: 12px;
  font-weight: normal;
  box-sizing: border-box;
}

<span
    class="c0"
    color="green"
  >
    status label(outline)
  </span>
</DocumentFragment>
`;
